trade-finance-manager-api/test-helpers/with-validate-payload.tests.ts

```diff
@@ -0,0 +1,57 @@
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { NextFunction, Request, Response } from 'express';
+
+type ValidatePayloadTestCases = { aTestCase: () => unknown; description: string }[];
+
+/**
+ * This is a reusable test to allow for validating payloads
+ * It is intended to be used for any controller-level function that utilises
+ * `req`, `res` and `next` as parameters
+ * It allows for commonisation of test cases and commonisation of tests
+ */
```

Comment:
Do we intend to use this for other non-SSO controllers?

Reply:
Yep -- Anywhere where we want to test validate payload middleware

Reply:
(the point is here is that we should always throw a bad request so we can throw in a bunch of tests for this and its very reusable)

Reply:
I can't actually see this being used anywhere in this branch?

Concern here is that this seems to be a new pattern / different to other existing payload validations in controllers and unless we create a ticket in the backlog, we could have different patterns for doing the same thing

Reply:
Ah I havent actually added it to a test -- I have done now. It's a new testing pattern that allows us to test validation through using existing schema tests -- In `users.routes.test` we now use it -- and it allows us to test our validation is correctly being called without mocks
```
  user routes
    upsertTfmUserFromEntraIdUser
      when validating the payload
        √ should respond with a '400' if the oid is missing (5 ms)  
        √ should respond with a '400' if the verified primary email is missing (1 ms)
        √ should respond with a '400' if the verified secondary email is missing (1 ms)
        √ should respond with a '400' if the given name is missing  
        √ should respond with a '400' if the family name is missing 
        √ should respond with a '400' if the roles are missing      
        √ should respond with a '400' if the oid is not a string    
        √ should respond with a '400' if the verify primary email array is empty (1 ms)
        √ should respond with a '400' if the verify primary email is not a string array (1 ms)
        √ should respond with a '400' if the verify primary email is not an array
        √ should respond with a '400' if the verify secondary email is not a string array
        √ should respond with a '400' if the verify secondary email is not an array
        √ should respond with a '400' if the given name is not a string
        √ should respond with a '400' if the family name is not a string
        √ should respond with a '400' if the roles are not an array of user roles (1 ms)
        √ should respond with a '400' if the roles are not an array (1 ms)
        √ should respond with a '400' if the object is empty        
        √ should return a success status if a complete valid payload is present
        √ should return a success status if the verified secondary email array is empty
        √ should return a success status if the roles array is empty (1 ms)
        √ should return a success status if there is an extra field 

// The below is not part of the validation tests
      when the request body is valid
        √ calls the user controller
        when the upsert is unsuccessful
          when the error is an api error
            √ returns the api error with expected formatting (1 ms) 
          when the error is not an api error
            √ passes the error to further middleware (1 ms)
        when the upsert is successful                               
          √ returns the result of the user controller      
```

Reply:
In this case -- this function is checking the `ENTRA_ID_USER_SCHEMA` and the errors/status returned